test/framework/src/test/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertionsTests.java

```diff
@@ -188,4 +200,49 @@ public void testAssertXContentEquivalentErrors() throws IOException {
             assertThat(error.getMessage(), containsString("expected [1] more entries"));
         }
     }
+
+    public void testAssertBlocked() {
+        {
+            List<DefaultShardOperationFailedException> shardFailures = new ArrayList<>();
+            Set<ClusterBlock> clusterBlocks = new HashSet<>();
+            clusterBlocks.add(IndexMetaData.INDEX_READ_ONLY_BLOCK);
```

Comment:
Suggest to use:
```
Set<ClusterBLock> clusterBlocks = Set.of(IndexMetaData.INDEX_READ_ONLY_BLOCK);
```
This ensures master is shorter, in backport to 7.x, we can do either `Collections.singleton` or `new HashSet<>(Arrays.asList(XX,YY))` (I can take care of that part).